### PR TITLE
Changelogs: Standardize sections

### PR DESCRIPTION
**Pull request by sirreal**
## What?

Improve existing changelog entries to adhere to a more standard format. I've changed all similar names the seem clear to be:

- Breaking Changes
- Bug Fixes
- Deprecations
- Enhancements
- New Features
- Code Quality -> Internal
- Improvements -> Enhancements
- New API -> New Features
- Polish -> Internal
- Refactor -> Internal
- Requirements -> Breaking Change
- Backward Compatibility -> Bug Fixes

[I've updated the documentation as well.](https://github.com/WordPress/gutenberg/blob/fix/changelog-standard-sections/packages/README.md#maintaining-changelogs)

I've also updated the root changelog and the changelog script to use "first-time contributors" (hyphenated) instead of "first time contributors." [I understand this is correct](https://www.merriam-webster.com/dictionary/first-time), where "first-time" is and adjective.

Should more things be moved to standard names?

- [ ] Documentation
- [ ] Experimental
- [ ] Performance -> ?
- [ ] Security (? maybe there's a better section - these were updating vulnerable dependencies)

## Why?

We expect a relatively standard format but nothing enforces it. Using regular sections in CHANGELOGs helps releases go smoothly and keeps changelogs clear.

```sh
grep -ih '^### ' packages/*/CHANGELOG.md | sort | uniq -c
```
```diff
diff --git before/trunk after/branch
index 5864d8ca18e..82de4dec05d 100644
--- before/trunk
+++ after/branch
@@ -1,33 +1,13 @@
       1 ### Backward Compatibility
-    126 ### Breaking Change
-    315 ### Breaking Changes
-      7 ### Breaking change
-      8 ### Breaking changes
-      1 ### Bug
-    204 ### Bug Fix
-    110 ### Bug Fixes
-     10 ### Bug fix
-      6 ### Bug fixes
+    458 ### Breaking Changes
+    331 ### Bug Fixes
       2 ### Code Quality
-     13 ### Deprecation
-     41 ### Deprecations
+     54 ### Deprecations
      11 ### Documentation
-     73 ### Enhancement
-    131 ### Enhancements
+    214 ### Enhancements
      29 ### Experimental
-      3 ### Feature
-      1 ### Features
-     10 ### Improvements
       2 ### Initial Release
-    127 ### Internal
-      4 ### New API
-    112 ### New Feature
-    159 ### New Features
-     23 ### New feature
-      8 ### New features
+    163 ### Internal
+    310 ### New Features
       1 ### Performance
-     34 ### Polish
-      2 ### Refactor
-      3 ### Requirements
-      1 ### Security
-      1 ### Security Fix
+      2 ### Security
```

### CODE REVIEW

**Comment by gziolo** [comment]
> Should more things be moved to standard names?

I would bring closer the headlines with available labels for PRs and issues. @priethor, might be the best person to help you bring clarity here.

I have a larger change in the release flow that would benefit from the changes you started here.

File: bin/plugin/commands/common.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ function calculateVersionBumpFromChangelog(
 	let changesDetected = false;
 	let versionBump = null;
 	for ( const line of lines ) {
-		const lineNormalized = line.toLowerCase().trimLeft();
+		const lineNormalized = line.toLowerCase().trimStart();
```
Comment (sirreal) [inline bin/plugin/commands/common.js:49]: This is replacing a deprecated `String` method (alias) with its recommended replacement.

**Comment by sirreal** [comment]
I think we should land this, we can always tweak the section names later. The more standard things are the less likely they are to drift.

**Review by t-hamano — COMMENTED** [review]
I feel like this PR is okay to ship, but what about correcting the incorrect heading level as well?

In some places it appears that a level 2 heading is being used where a level 3 heading is expected.

```
$ grep -ih '^## [^0-9]' packages/*/CHANGELOG.md | grep -iv '## unreleased' | sort | uniq -c
      1 ## Breaking Change
      2 ## Breaking Changes
      1 ## Bug Fixes
      1 ## Deprecations
      3 ## Enhancements
      2 ## New Feature
      1 ## New features
```

**Comment by sirreal** [comment]
> I feel like this PR is okay to ship, but what about correcting the incorrect heading level as well?

Good spot, I've cleaned up the wrong levels in package changelogs.

This also lead me to the root changelog.txt file where I did a fair amount of cleanup and standardization. It would be nice to change that file to an `.md` file and run autoformatting on it 🤔

File: changelog.txt [hunk 2]
```diff
@@ -36319,8 +36331,10 @@ Fix action GitHub action workflow YAML syntax errors. ([23844](https://github.co
 
 = 7.8.0 =
 
-## Enhancements
-​
+# Changelog
```
Comment (t-hamano) [inline changelog.txt]:
```suggestion
## Changelog
```

File: changelog.txt [hunk 3]
```diff
@@ -1985,9 +1991,9 @@ The following contributors merged PRs in this release:
 - Distraction free: Remove unwanted space from string. ([60108](https://github.com/WordPress/gutenberg/pull/60108))
 
 
-## First time contributors
+## First-time contributors
 
-The following PRs were merged by first time contributors 🎉 :
+The following PRs were merged by first-time contributors 🎉 :
```
Comment (t-hamano) [inline changelog.txt:1996]:
```suggestion
The following PRs were merged by first-time contributors:
```

File: bin/plugin/commands/changelog.js [hunk 4]
```diff
@@ -961,9 +961,9 @@ function getContributorProps( pullRequests ) {
 	}
 
 	return (
-		'## First time contributors' +
+		'## First-time contributors' +
 		'\n\n' +
-		'The following PRs were merged by first time contributors:' +
+		'The following PRs were merged by first-time contributors:' +
```
Comment (t-hamano) [inline bin/plugin/commands/changelog.js:966]: We may also need to update the unit test [snapshot](https://github.com/WordPress/gutenberg/blob/5686e54f36447ed0d38016e15a633ad4a5a05a89/bin/plugin/commands/test/__snapshots__/changelog.js.snap#L287).

```
$ npm run test:unit bin/plugin/commands/test/changelog.js

...
● getContributorProps › verify that the contributors props are properly formatted

    expect(received).toMatchSnapshot()

    Snapshot name: `getContributorProps verify that the contributors props are properly formatted 1`

    - Snapshot  - 2
    + Received  + 2

    - ## First time contributors
    + ## First-time contributors

    - The following PRs were merged by first time contributors:
    + The following PRs were merged by first-time contributors:
...
```

**Review by t-hamano — APPROVED** [review]
I left a few suggestions at the end, all of which appear to be formatted correctly.

File: bin/plugin/commands/changelog.js (hunk 4 above)
Comment (sirreal) [reply bin/plugin/commands/changelog.js:966]: Done, thanks.

**Pull request merged**